libs/newsletters-data-client/src/lib/newsletter-storage/s3-newsletter-storage.ts

```diff
@@ -282,6 +281,46 @@ export class S3NewsletterStorage implements NewsletterStorage {
 		}
 	}
 
+	async replace(
+		listId: number,
+		newsletter: NewsletterData,
+		user: UserProfile,
+	): Promise<
+		| SuccessfulStorageResponse<NewsletterDataWithoutMeta>
+		| UnsuccessfulStorageResponse
+	> {
+		const newsletterToUpdate = await this.fetchNewsletter(listId);
+		if (!newsletterToUpdate) {
+			return {
+				ok: false,
+				message: `failed to read newsletter with id ${listId}`,
+			};
+		}
+		const { identityName } = newsletterToUpdate;
+		const updatedNewsletter: NewsletterDataWithMeta = {
```

Comment:
As with the in memory version - maybe throw an error if `newsletter.identityName !== newsletterToUpdate.identityName || newsletter.listId !== newsletterToUpdate.listId`.